client/src/components/LoginSession/LoginSessionContainer.js

```diff
@@ -1,80 +1,65 @@
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import { connect } from 'react-redux';
 import backend from 'lib/Backend';
 import Config from 'lib/Config'; // eslint-disable-line
 import LoginSession from './LoginSession';
 import { success, error } from 'state/toasts/ToastsActions';
+import i18n from 'i18n';
 
-// Handle communication with  server on logout and toast notifications via redux
-// Using a class component rather than function component so that LoginSessionContainer-test.js
-// can use wrapper.instance() to test logout().
-// .instance() does not work on stateless functional components
-// https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/instance.html
-class LoginSessionContainer extends Component {
-    //
-    constructor(props) {
-        super(props);
-        this.createEndpoint = this.createEndpoint.bind(this);
-        this.logout = this.logout.bind(this);
-    }
-
-    componentWillMount() {
-        this.setState({
-            complete: false,
-            failed: false,
-            submitting: false
-        });
-    }
+function createEndpoint(logOutEndpoint) {
+    return backend.createEndpointFetcher({
+        url: `${logOutEndpoint}/:id`,
+        method: 'delete',
+        payloadSchema: {
+            id: { urlReplacement: ':id', remove: true },
+            SecurityID: { querystring: true }
+        }
+    });
+}
 
-    createEndpoint() {
-        return backend.createEndpointFetcher({
-            url: `${this.props.LogOutEndpoint}/:id`,
-            method: 'delete',
-            payloadSchema: {
-                id: { urlReplacement: ':id', remove: true },
-                SecurityID: { querystring: true }
-            }
-        });
-    }
+/**
+ * Handle communication with  server on logout and toast notifications via redux.
+ * @param props
+ */
+function LoginSessionContainer(props) {
+    const [complete, setComplete] = useState(false);
+    const [failed, setFailed] = useState(false);
+    const [submitting, setSubmitting] = useState(false);
```

Comment:
I like this way of splitting the each piece of state into a variable, rather than a single object with keys